src/Castle.Core/DynamicProxy/ModuleScope.cs

```diff
@@ -133,8 +143,21 @@ public ModuleScope(bool savePhysicalAssembly, bool disableSignedModule, string s
 		internal ModuleScope(bool savePhysicalAssembly, bool disableSignedModule, INamingScope namingScope,
 		                     string strongAssemblyName, string strongModulePath,
 		                     string weakAssemblyName, string weakModulePath)
+			: this(
+#if FEATURE_ASSEMBLYBUILDER_SAVE
+				  assemblyBuilderAccess: savePhysicalAssembly ? AssemblyBuilderAccess.RunAndSave : AssemblyBuilderAccess.Run,
+#else
+				  assemblyBuilderAccess: AssemblyBuilderAccess.Run,
+#endif
```

Comment:
This is the code location where we could add additional logic to default to `RunAndCollect` instead of `Run` depending on the current `AssemblyLoadContext`.

Reply:
As mentioned in https://github.com/castleproject/Core/issues/473#issuecomment-2390642865, auto-detection logic may indeed not be the most pressing feature, so I am omitting any such logic, at least for now.